### ISSUE TITLE
Segment with GPU - Statistics is not getting calculated. 

### ISSUE THREAD

**Issue by laulenne** [ultralytics/ultralytics#14748]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

I am runnning the following configuration:
CUDA 12.4
torch: 2.4.0+cu124
torchvision: 0.19.0+cu124
ultralytics: 8.2.67
Graphic interface NVIDIA L Quatro T1000

When running the training with the following configuration 
```python
 results = model.train(data="segmentdata.yaml", 
                        epochs=100, 
                        imgsz=640                       
 ) 

```
I get the following issue: only the segment_loss is getting caculate, the box are Nan.
![image](https://github.com/user-attachments/assets/85a5f88a-e158-4a60-981a-5b0964dcf676)

When i trained the model with the following configuration 
```python
results = model.train(data="segmentdata.yaml", 
                        epochs=100, 
                        imgsz=640, 
                        amp=False,
                        lr0=0.001,
                        )  
```
The loss get caculated bu the aMP are not getting calculated. 
![image](https://github.com/user-attachments/assets/32e480fa-a2a3-4091-aad2-efae0a2fe927)

### Environment

CUDA 12.4
torch: 2.4.0+cu124
torchvision: 0.19.0+cu124
ultralytics: 8.2.67
Graphic interface NVIDIA L Quatro T1000

### Minimal Reproducible Example

When running the training with the following configuration 
```python
 results = model.train(data="segmentdata.yaml", 
                        epochs=100, 
                        imgsz=640                       
 ) 
```

I get the following issue: only the segment_loss is getting caculate, the box are Nan.
![image](https://github.com/user-attachments/assets/85a5f88a-e158-4a60-981a-5b0964dcf676)

When i trained the model with the following configuration 
```python
results = model.train(data="segmentdata.yaml", 
                        epochs=100, 
                        imgsz=640, 
                        amp=False,
                        lr0=0.001,
                        )  
```
The loss get caculated bu the aMP are not getting calculated. 
![image](https://github.com/user-attachments/assets/32e480fa-a2a3-4091-aad2-efae0a2fe927)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Y-T-G** [comment]
Your GPU probably doesn't support amp.

**Comment by laulenne** [comment]
my GPU supports AMP. 
>>> major, minor = torch.cuda.get_device_capability()
>>> major
7

**Comment by Y-T-G** [comment]
Some GPUs support AMP, but it's broken. Like GTX 16xx series. It also has nan issue with amp.

**Comment by Y-T-G** [comment]
All these devices probably have issue with float16:

https://github.com/easydiffusion/easydiffusion/blob/716f30fecb3497644eb1033cc63f965a19c3f871/ui/easydiffusion/device_manager.py#L158
